gordo_components/builder/build_model.py

```diff
@@ -486,7 +486,7 @@ def check_cache(self, model_register_dir: Union[os.PathLike, str]):
             )
             return None
 
-    def provide_saved_model(
+    def build_with_cache(
```

Comment:
Hmm. Do you feel this is explaining the function good enough? :thinking:

Reply:
:man_shrugging: Slightly better than previous? Do you have a suggestion? :pray:

Reply:
Yes, I think it is better than the previous, but I am uncertain as it is a "if this else that" kind of function. But as I have no better to come up with, please ignore my nitpicking. :pick:

Reply:
Ya, I agree. It feels like this outer check in the CLI of `if evaluation_config["cv_mode"] == ...` should be held inside of the `ModelBuilder`. But maybe that's ok actually? I don't really know. I implemented it as such inside the builder, but it looked ugly. Sooo, I've switched it back and will think more. Otherwise we can always do an additional (smaller) refactor after this is in. :question:

Reply:
Yeah, it is related to the the model building as it checks what to build/skip. But I think it is fine to do a smaller refactor later if we need it. It is quite a rewrite itself, so doing CLI in it's own is perhaps fine.